src/init/starship.ps1

```diff
@@ -142,22 +146,50 @@ $null = New-Module starship {
         $arguments += "--status=$($lastExitCodeForPrompt)"
 
         # Invoke Starship
-        $promptText = if ($script:TransientPrompt) {
+        $promptText, $rpromptText = $null, $null
+        if ($script:TransientPrompt) {
             $script:TransientPrompt = $false
-            if (Test-Path function:Invoke-Starship-TransientFunction) {
+            $promptText = if (Test-Path function:Invoke-Starship-TransientFunction) {
                 Invoke-Starship-TransientFunction
             } else {
                 "$([char]0x1B)[1;32m❯$([char]0x1B)[0m "
             }
+            $rpromptText = if (Test-Path function:Invoke-Starship-TransientRightFunction) {
+                Invoke-Starship-TransientRightFunction
+            } else {
+                ""
+            }
         } else {
-            Invoke-Native -Executable ::STARSHIP:: -Arguments $arguments
+            $lproc = [System.Diagnostics.Process]::Start((Build-ProcessInfo -Executable ::STARSHIP:: -Arguments $arguments))
+            $rproc = [System.Diagnostics.Process]::Start((Build-ProcessInfo -Executable ::STARSHIP:: -Arguments ($arguments + "--right")))
+            $lsout = $lproc.StandardOutput.ReadToEndAsync()
+            $rsout = $rproc.StandardOutput.ReadToEndAsync()
+            $lserr = $lproc.StandardError.ReadToEndAsync()
+            $rserr = $rproc.StandardError.ReadToEndAsync()
+            [System.Threading.Tasks.Task]::WaitAll(@($lsout, $rsout, $lserr, $rserr))
+            if ($lserr.Result.Trim() -ne '') { $host.ui.WriteErrorLine($lserr.Result) }
+            if ($rserr.Result.Trim() -ne '') { $host.ui.WriteErrorLine($rserr.Result) }
+            $promptText  = $lsout.Result;
+            $rpromptText = $rsout.Result;
         }
 
+        $promptLines = $promptText.Split("`n")
+        # Calculate offset for printing on right side:
+        # offset = (console width) + (length of ANSI sequences on right_prompt) - (length of just text in last line of left_prompt)
+        $rpromptOffset = $Host.UI.RawUI.WindowSize.Width + ($rpromptText.Length - ($rpromptText -replace $ansiRegex).Length) - ($promptLines[-1] -replace $ansiRegex).Length
+
```

Comment:
I probably should have tested this earlier, but I just noticed that if the last prompt line contains an emoji or any double-width (or any non-single-width character), this calculation will break.
Does PowerShell have any nice Unicode-aware string width functions via .NET that we could use instead?

Reply:
I tried some emojis and double-width characters (◌️, 😂, 😢) and they seem to be working well.

![image](https://user-images.githubusercontent.com/46838874/211205388-fb880948-656d-4b47-81e5-3d7d59389354.png)

![image](https://user-images.githubusercontent.com/46838874/211205510-936c4671-42c0-46c5-8c04-b0d9ef62f78c.png)

Do you have any characters I can test?

Reply:
`❌` from the status triggered a rendering issue in iTerm for me:
<img width="85" alt="image" src="https://user-images.githubusercontent.com/835177/211205683-acdbe1a6-506c-4504-84b1-38fd9113de48.png">

You can try this:
 ```
[character]
success_symbol = "❌❌❌❌❌❌❌❌❌❌❌"
```

Reply:
Hmm, I can repro the issue using this symbol. I have tried searching online, but I can't find a proper method to find out the number of terminal boxes a character occupies, in C#. There are ways to find out code point or grapheme lengths, but they all mean different things 😢

Reply:
Starship itself uses the maximum code point width per grapheme to approximate character widths, so using something along those line should be fine.

Reply:
Could you point me where the code for that lives? I can try translating it to PowerShell.

Reply:
It's defined in `print.rs`:
https://github.com/starship/starship/blob/master/src/print.rs#L21-L48

Reply:
Okay, I'll admit I have no idea what's happening there 😅. It seems to be using a third-party library and its members.

Could we handle this in a new issue/PR?

Reply:
> Could we handle this in a new issue/PR?

I would rather not, we're already getting quite a few issues because `$fill` doesn't always calculate widths as expected.
At least this should use some kind of string width function like `wcswidth` verbatim.

Reply:
Unfortunately Windows or .NET doesn't have a `wcswidth` equivalent. Most portable implementations of `wcswidth` I found online use lookup tables of code points. Even the unicode-width Rust library (used in `print.rs`) has a table. Implementing that in the PowerShell init script does not sound very feasible to me.

Reply:
In that case, I would rather go with #4208. I'll try to push the `--profile` PR forward and rendering custom target pairs after that to allow for transient rprompts in pwsh. Hopefully this can work out without a breaking change to transient prompt configuration.

Reply:
Okay